### PR TITLE
Fix debug_traceTransaction endpoint

### PR DESCRIPTION
**Pull request by ToniRamirezM**
### What does this PR do?

Fixes debug_traceTransaction endpoint.

Adds a commented test to compare response between geth and zkevm-node used during development of this fix.

### Reviewers

Main reviewers:

- @tclemos 
- @KonradIT

### CODE REVIEW

**Comment by KonradIT** [comment]
Is this still in progress?

**Comment by ToniRamirezM** [comment]
No, it is ready for review

File: jsonrpc/debug_test.go [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+func TestMain(m *testing.M) {
+	// initOrResetDB()
+
+	stateDb, err = db.NewSQLDB(stateDBCfg)
+	if err != nil {
+		panic(err)
+	}
+	defer stateDb.Close()
+
+	zkProverURI := testutils.GetEnv("ZKPROVER_URI", "34.245.216.26")
```
Comment (KonradIT) [inline jsonrpc/debug_test.go:71]:
```suggestion
	zkProverURI := testutils.GetEnv("ZKPROVER_URI", "localhost")
```

**Review by KonradIT — APPROVED** [review]

**Comment by KonradIT** [comment]
- Given the code calls the two networks directly, I'd put it in `test/e2e/` like the jsonrpc_test for discrepancy discovery. The `jsonrpc_test.go` in `jsonrpc/` uses mocking (unit test)
- Switch to a for loop and use `shared.go:networks` slice for iterating over the 2 networks

**Comment by KonradIT** [comment]
How I'd do the test as part of `e2e/jsonrpc_test.go`:

```go
func Test_Debug(t *testing.T) {
	if testing.Short() {
		t.Skip()
	}
	Setup()
	defer Teardown()

	var senderAddress = common.HexToAddress("0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266")
	var senderPvtKey = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"

	for _, network := range networks {
		log.Infof("Network %s", network.Name)

		client, err := ethclient.Dial(network.URL)
		require.NoError(t, err)

		// ChainId()
		chainId, err := client.ChainID(ctx)
		require.NoError(t, err)
		require.Equal(t, network.ChainID, chainId.Uint64())

		nonce, err := client.PendingNonceAt(ctx, senderAddress)
		require.NoError(t, err)
		log.Debugf("nonce = %v", nonce)

		privateKey, err := crypto.HexToECDSA(strings.TrimPrefix(senderPvtKey, "0x"))
		require.NoError(t, err)

		auth, err := bind.NewKeyedTransactorWithChainID(privateKey, chainId)
		require.NoError(t, err)

		balance, err := client.BalanceAt(context.Background(), senderAddress, nil)
		require.NoError(t, err)
		require.NotEqual(t, "0", balance.String())

		// Deploy SC
		scRevertByteCode, err := testutils.ReadBytecode("Revert2/Revert2.bin")
		require.NoError(t, err)

		gasPrice, err := client.SuggestGasPrice(ctx)
		require.NoError(t, err)

		// Deploy revert.sol
		tx := types.NewTx(&types.LegacyTx{
			Nonce:    nonce,
			To:       nil,
			Value:    new(big.Int),
			Gas:      uint64(4000000),
			GasPrice: gasPrice,
			Data:     common.Hex2Bytes(scRevertByteCode),
		})

		signedTx, err := auth.Signer(auth.From, tx)
		require.NoError(t, err)

		err = client.SendTransaction(ctx, signedTx)
		require.NoError(t, err)
		log.Debugf("txHash = %v", signedTx.Hash())

		// Wait fot the transaction to be mined
		err = operations.WaitTxToBeMined(ctx, client, signedTx, operations.DefaultTimeoutTxToBeMined)
		require.NoError(t, err)
	
		_, err = debugTransaction(t, network.URL, signedTx.Hash().String())
		require.NoError(t, err)
	}
}
```

**Comment by ToniRamirezM** [comment]
An issue to move the test can be added. We should merge the fix as soon as possible.

**Comment by KonradIT** [comment]
Ok merge it, I approved earlier.

File: jsonrpc/debug_test.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
+package jsonrpc_test
```
Comment (tclemos) [inline jsonrpc/debug_test.go:1]: we should have this test on `e2e` tests

**Review by tclemos — APPROVED** [review]

File: jsonrpc/debug_test.go (hunk 2 above)
Comment (KonradIT) [reply jsonrpc/debug_test.go:1]: Issue already created yep: https://github.com/0xPolygonHermez/zkevm-node/issues/1343 plus I supplied code that can be used in a previous comment.

**Pull request merged**